File: Frontend/src/app/models/maker/tax.ts
import { TaxFile } from "./tax-file";

export class Tax {
  id?: number;
  mainGuid?: String
  from_?: number;
  sendTo_?: number;
  taxCategory?: number;
  noOfEmployee?: number;
  taxableAmount?: number;
  taxWithHold?: number;
  incometaxPoolAmount?: number;
  graduatetaxPool?: number;
  graduatetaxPool1?: number;
  graduaTotBasSalary?: number;
  graduateTotaEmployee?: number;
  graduatetaxWithHold?: number;
  taxCategoryList?: string;
  maker_name?: string;
  maker_date?: Date;
  checker_name?: string;
  checked_Date?: Date;
  updated_user_name?: string;
  updated_event_date?: Date;
  from_List?: string;
  sendTo_List?: string;
  Category_List?: String;
  FileDetail?: number;
  status?: number;
  taxFile?: TaxFile[] = []
  previouseTaxFile?: TaxFile[] = []

  remark?: string
  reference_number?: string
  initiator_branch?: string;   // The branch that initiated the tax
  destination_branch?: string; // The branch that receives/sends to
  taxType?: string; // tax category type
  user_id?: number;
  rejector_checker_id?: number;
  checker_rejected_date?: Date;
  checker_rejected_reason?: string;
  rejector_approver_id?: number;
  approver_rejected_reason?: string;
  approver_rejected_date?: Date;
  approver_name?: string;
  approved_date?: Date;
  isFileEdited?: boolean = false;
  drafted_date?: Date;
  maker_id ?: number

}

File: Backend/TaxManagementSystem/src/main/java/com/tms/Maker/entity/Tax.java
package com.tms.Maker.entity;

import java.sql.Date;
import java.util.List;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class Tax {
    private Long id;
    private String mainGuid;
    private int from_;
    private int sendTo_;
    private int taxCategory;
    private int noOfEmployee;
    private float taxableAmount;
    private float taxWithHold;
    private float incometaxPoolAmount;
    private float graduatetaxPool;
    private float graduaTotBasSalary;
    private int graduateTotaEmployee;
    private float graduatetaxWithHold;
    private String taxCategoryList;
    private String Remark;
    private String maker_name;
    private Date maker_date;
    private String checker_name;
    private Date checked_Date;
    private String updated_user_name;
    private Date updated_event_date;
    private String from_List;
    private String sendTo_List;
    private String Category_List;
    private int status;
    private List<TaxFile> taxFile;
    private String remark;
    private String initiator_branch;
    private String destination_branch;
    private String taxType;
    private Long user_id;
    private Date approver_rejected_date;
    private String approver_rejected_reason;
    private Long rejector_approver_id;
    private String checker_rejected_reason;
    private Date checker_rejected_date;
    private Long rejector_checker_id;
    private String approver_name;
    private Date approved_date;
    private String reference_number;

    private Boolean isFileEdited;
    private List<TaxFile> previouseTaxFile;

    private Long maker_id;
    private Date drafted_date;
    private String fileExsistance;

}